videoclip.lua

```diff
@@ -431,6 +448,69 @@ function main_menu:update()
     self:overlay_draw(osd:get_text())
 end
 
+function main_menu:upload_catbox()
+    local endpoint = config.litterbox and 'https://litterbox.catbox.moe/resources/internals/api.php' or 'https://catbox.moe/user/api.php'
+
+    encoder.create_clip('video',
+        function(outfile)
+            notify("Uploading to " .. (config.litterbox and "litterbox.catbox.moe..." or "catbox.moe..."), "info", 9999)
+
+            -- This uses cURL to send a request to the cat-/litterbox API.
+            -- cURL is included on Windows 10 and up, most Linux distributions and macOS.
+
+            local r = mp.command_native({ -- This is technically blocking, but I don't think it has any real consequences ..?
+                name = 'subprocess',
+                playback_only = false,
+                capture_stdout = true,
+                capture_stderr = true,
+                args = {
+                    os_type == 'window' and 'curl.exe' or 'curl', '-s',
```

Comment:
`os_type == 'window'` is never `true`. Calling `curl` should be sufficient on all systems. As far as I know, it shouldn't be necessary to append `.exe` to the executable name, even on Windows.